### ISSUE TITLE
[Twitch] A new "Commercial Break" segment appears when an ad is played

### ISSUE THREAD

**Issue by David-Eilers** [streamlink/streamlink#3119]
## Bug Report

- [x] This is a bug report and I have read the contribution guidelines.

### Description

It seems that twitch has implement a segment of video which plays when a ad would normally have been show.
It shows the twitch logo in the top center of the screen with the text "Commercial Break in Progress" below it with a timer in the bottom right corner, this is all on top of a "twitch purple" background.

The elapsed time in VLC is set to zero when the segment starts playing and is reverted to the original value once the break is over.
To me this hints at a some sort of separate video source rather then the segment being spliced in the actual stream

### Expected / Actual behavior

Expected: a uninterrupted livestream of the user i'm watching as long as the stream is live.

Actual Behavior: The livestream is interrupted by a purple screen which says "Commercial Break In Progress" with a timer in the bottom left when an ad would usually be playing, when opening the livestream or during the stream when triggered by the streamer 

### Reproduction steps / Explicit stream URLs to test

1. ... open a stream which has an pre-roll ad or which manually triggers twitch ads during their broadcast. 
1a .. observe the commercial break in progress segment after opening the stream
1b. .. wait for an ad to play and then observe the commercial break in progress segment

Urls on which i witnessed this new segment

- twitch.tv/gamesdonequick
- twitch.tv/zisteau
- twitch.tv/arkas
- twitch.tv/cohhcarnage
- and any other random twitch streamer

### Log output

```
PS C:\Users\david> streamlink.exe --loglevel debug twitch.tv/cohhcarnage best
[cli][debug] OS:         Windows 10
[cli][debug] Python:     3.7.6
[cli][debug] Streamlink: 1.5.0
[cli][debug] Requests(2.21.0), Socks(1.6.7), Websocket(0.56.0)
[cli][info] Found matching plugin twitch for URL twitch.tv/cohhcarnage
[plugin.twitch][debug] Getting live HLS streams for cohhcarnage
[utils.l10n][debug] Language code: nl_NL
[cli][info] Available streams: audio_only, 160p (worst), 360p, 480p, 720p, 720p60, 1080p60 (best)
[cli][info] Opening stream: 1080p60 (hls)
[stream.hls][debug] Reloading playlist
[stream.hls][debug] First Sequence: 0; Last Sequence: 2
[stream.hls][debug] Start offset: 0; Duration: None; Start Sequence: 0; End Sequence: None
[stream.hls][debug] Adding segment 0 to queue
[cli][debug] Pre-buffering 8192 bytes
[stream.hls][debug] Adding segment 1 to queue
[stream.hls][debug] Adding segment 2 to queue
[stream.hls][debug] Download of segment 0 complete
[cli][info] Starting player: "C:\Program Files\VideoLAN\VLC\vlc.exe"
[cli.output][debug] Opening subprocess: "C:\Program Files\VideoLAN\VLC\vlc.exe" --input-title-format twitch.tv/cohhcarnage -
[stream.hls][debug] Download of segment 1 complete
[stream.hls][debug] Download of segment 2 complete
[cli][debug] Writing stream to output
[stream.hls][debug] Reloading playlist
[stream.hls][debug] Adding segment 3 to queue
[stream.hls][debug] Adding segment 4 to queue
[stream.hls][debug] Download of segment 3 complete
[stream.hls][debug] Download of segment 4 complete
[stream.hls][debug] Reloading playlist
[stream.hls][debug] Adding segment 5 to queue
[stream.hls][debug] Adding segment 6 to queue
[stream.hls][debug] Adding segment 7 to queue
[stream.hls][debug] Download of segment 5 complete
[stream.hls][debug] Download of segment 6 complete
[stream.hls][debug] Download of segment 7 complete
[stream.hls][debug] Reloading playlist
[stream.hls][debug] Adding segment 8 to queue
[stream.hls][debug] Download of segment 8 complete
[stream.hls][debug] Reloading playlist
[stream.hls][debug] Adding segment 9 to queue
[stream.hls][debug] Download of segment 9 complete
[stream.hls][debug] Reloading playlist
[stream.hls][debug] Adding segment 10 to queue
[stream.hls][debug] Adding segment 11 to queue
[stream.hls][debug] Download of segment 10 complete
[stream.hls][debug] Download of segment 11 complete
[stream.hls][debug] Reloading playlist
[stream.hls][debug] Adding segment 12 to queue
[stream.hls][debug] Download of segment 12 complete
[stream.hls][debug] Reloading playlist
[stream.hls][debug] Adding segment 13 to queue
[stream.hls][debug] Adding segment 14 to queue
[stream.hls][debug] Download of segment 13 complete
[stream.hls][debug] Download of segment 14 complete
[stream.hls][debug] Reloading playlist
[stream.hls][debug] Adding segment 15 to queue
[stream.hls][debug] Download of segment 15 complete
[stream.hls][debug] Reloading playlist
[stream.hls][debug] Adding segment 16 to queue
[stream.hls][debug] Download of segment 16 complete
[stream.hls][debug] Reloading playlist
[stream.hls][debug] Adding segment 17 to queue
[stream.hls][debug] Download of segment 17 complete
[cli][info] Player closed
[stream.segmented][debug] Closing worker thread
[stream.segmented][debug] Closing writer thread
[stream.segmented][debug] Closing writer thread
[cli][info] Stream ended
[cli][info] Closing currently open stream...
```

### Additional comments, screenshots, etc.

Log posted above is for the whole commercial break segment plus around 15 seconds of the actual stream

Screenshot of the actual segment:
![twitch-commercial](https://user-images.githubusercontent.com/3015646/90697683-3ec45680-e27f-11ea-8302-ea1b5e3685bb.png)

[Love Streamlink? Please consider supporting our collective. Thanks!](https://opencollective.com/streamlink/donate)

**Comment by bastimeyer** [comment]
https://streamlink.github.io/cli.html#cmdoption-twitch-disable-ads

**Comment by David-Eilers** [comment]
ok, i will close the isssue as it appears this is a known problem and there is a solution. my sincere apologies.

Just out of curiosity, is there a reason why this a cmd option and not default behavior?

**Comment by bastimeyer** [comment]
> is there a reason why this a cmd option and not default behavior?

Because when the parameter is set, it filters the HLS playlist's segments, which is not default behavior of Streamlink, as it's just a simple HLS client. This is additional logic. The intention of Streamlink is not to prevent ads. It's up to the user to enable this.

If you want this to be your system's default behavior, then you can add it to the global config file or create a config file for the twitch plugin.

**Comment by David-Eilers** [comment]
I see, i will change my script to use the option. thanks for the clarification.

**Comment by farid-fari** [comment]
Do we know why this is new? Why did it never occur for me until today?

**Comment by bastimeyer** [comment]
This is nothing new. Twitch has implemented and run embedded ads several times now, once even breaking our old implementation for filtering out ads which had to be rewritten. They don't always have embedded ads active, which is the reason why you've not been seeing this recently.

**Comment by iplaywithdrums** [comment]
I have already set the twitch disable ads option, and it has never shown ads to me before today.

Something has changed today, which is why you are seeing an influx of issues relating to the same topic.

**Comment by bastimeyer** [comment]
If it's not working for you, then post a debug log. It's working fine with the latest version on my system.

```
$ streamlink --loglevel debug --twitch-disable-ads twitch.tv/gamesdonequick best
[cli][debug] OS:         Linux-5.8.0-1-git-x86_64-with-glibc2.2.5
[cli][debug] Python:     3.8.5
[cli][debug] Streamlink: 1.5.0+10.g00fd4d2
[cli][debug] Requests(2.24.0), Socks(1.7.1), Websocket(0.56.0)
[cli][info] Found matching plugin twitch for URL twitch.tv/gamesdonequick
[cli][debug] Plugin specific arguments:
[cli][debug]  --twitch-disable-ads=True (disable_ads)
[plugin.twitch][debug] Getting live HLS streams for gamesdonequick
[utils.l10n][debug] Language code: en_US
[cli][info] Available streams: audio_only, 160p (worst), 360p, 480p, 720p, 720p60, 1080p60 (best)
[cli][info] Opening stream: 1080p60 (hls)
[plugin.twitch][info] Will skip ad segments
[stream.hls][debug] Reloading playlist
[plugin.twitch][info] Waiting for pre-roll ads to finish, be patient
[stream.hls][debug] First Sequence: 0; Last Sequence: 2
[stream.hls][debug] Start offset: 0; Duration: None; Start Sequence: 0; End Sequence: None
[stream.hls][debug] Adding segment 0 to queue
[cli][debug] Pre-buffering 8192 bytes
[stream.hls][debug] Adding segment 1 to queue
[stream.hls][debug] Adding segment 2 to queue
[stream.hls][debug] Reloading playlist
[stream.hls][debug] Adding segment 3 to queue
[stream.hls][debug] Reloading playlist
[stream.hls][debug] Adding segment 4 to queue
[stream.hls][debug] Adding segment 5 to queue
[stream.hls][debug] Adding segment 6 to queue
[stream.hls][debug] Reloading playlist
[stream.hls][debug] Adding segment 7 to queue
[stream.hls][debug] Adding segment 8 to queue
[stream.hls][debug] Download of segment 8 complete
[cli][info] Starting player: mpv
[cli.output][debug] Opening subprocess: mpv --title=twitch.tv/gamesdonequick --cache=no -
[cli][debug] Writing stream to output
[stream.hls][debug] Reloading playlist
[stream.hls][debug] Adding segment 9 to queue
[stream.hls][debug] Adding segment 10 to queue
[stream.hls][debug] Adding segment 11 to queue
[stream.hls][debug] Download of segment 9 complete
[cli][info] Player closed
[stream.segmented][debug] Closing worker thread
[stream.segmented][debug] Closing writer thread
[cli][info] Stream ended
[cli][info] Closing currently open stream...
```

> [plugin.twitch][info] Will skip ad segments
> [plugin.twitch][info] Waiting for pre-roll ads to finish, be patient
> [cli][info] Starting player: mpv

**Comment by Abalieno** [comment]
Yes, I've also opened an issue. This is new behavior, started a few hours ago in my experience. It's currently circumventing again the last fix.

**Comment by Abalieno** [comment]
Testing again. I've had to disable passthrough in the option files. 

The current behavior is that it waits the ads to end before starting the stream. I'll update in case I see ads interruptions again during the stream:

[plugin.twitch][info] Waiting for pre-roll ads to finish, be patient

**Comment by h1z1** [comment]
fwiw Twitch has done something else in the last 24-48 hours to where that "Commercial break in progress" preroll for all streams.  Funny thing is it also screws with the segment ids?

**Comment by gravyboat** [comment]
@h1z1 This isn't new at all. See the pinned issue in the repo, they've been turning this on and off for the last month plus.

**Comment by gravyboat** [comment]
I'm going to lock this, there's nothing more to discuss here that isn't already covered by the pinned issue.
